openapi/nexus.json

```diff
@@ -11822,6 +11822,17 @@
               }
             ]
           },
+          "mapped_fleet_roles": {
+            "description": "Mapping of which Fleet roles are conferred by each Silo role\n\nThe default is that no Fleet roles are conferred by any Silo roles unless there's a corresponding entry in this map.",
+            "type": "object",
+            "additionalProperties": {
+              "type": "array",
+              "items": {
+                "$ref": "#/components/schemas/FleetRole"
+              },
+              "uniqueItems": true
+            }
+          },
```

Comment:
Just noticed while integrating this into the console that this is probably wrong. I don't think Dropshot is handling `BTreeMap` correctly. I will look into it further.

Reply:
Hmmm, maybe it is right. I guess the keys can't be type-constrained, but we are correctly specifying the type of the values as an array of `FleetRole` where the items are unique. Ok!

https://swagger.io/docs/specification/data-models/dictionaries/

I will look at improving the TS client generator to handle this.

Reply:
If we wanted to be slightly more elaborate and more correct, instead of unconstrained string keys, maybe we could do the OpenAPI analogue of this TS type:

```ts
mapped_fleet_role: {
  viewer?: Set<FleetRole>,
  collaborator?: Set<FleetRole>,
  admin?: Set<FleetRole>,
}
```

i.e., a hard-coded set of keys, all of which are optional.